### DISCUSSION TITLE
Defining xarray schemas

### DISCUSSION THREAD

**Discussion by vilim** [pydata/xarray#5867] [Q&A] [answered]
Hello! Is there a way to define schemas for xarray datasets (e.g. that they always contain certain coordinates and DataArrays with elements of a certain types)? An extra plus would be if that would allow type hinting in IDEs/ipython

**Comment by keewis** [comment, marked as answer]
you might want to look at [`xarray-dataclasses`](https://github.com/astropenguin/xarray-dataclasses) for that (I never used it myself, though).

In general, the [ecosystem](https://xarray.pydata.org/en/latest/ecosystem.html) page is a good place to look for related libraries.

**Reply by vilim** [reply]
Thank you, that was exactly was I was looking for!

**Reply by benbovy** [reply]
This is also discussed in #1900.
